src/doc/rustdoc/src/documentation-tests.md

```diff
@@ -1,3 +1,239 @@
 # Documentation tests
 
-Coming soon!
\ No newline at end of file
+`rustdoc` supports executing your documentation examples as tests. This makes sure
+that your tests are up to date and working.
+
+The basic idea is this:
+
+```rust,ignore
+/// # Examples
+///
+/// ```
+/// let x = 5;
+/// ```
+```
+
+The triple backticks start and end code blocks. If this were in a file named `foo.rs`,
+running `rustdoc --test foo.rs` will extract this example, and then run it as a test.
+
+There's some subtlety though! Read on for more details.
+
+## Pre-processing examples
+
+In the example above, you'll note something strange: there's no `main`
+function! Forcing you to write `main` for every example, no matter how small,
+adds friction. So `rustdoc` processes your examples slightly before
+running them. Here's the full algorithm rustdoc uses to preprocess examples:
+
+1. Any leading `#![foo]` attributes are left intact as crate attributes.
+2. Some common `allow` attributes are inserted, including
+   `unused_variables`, `unused_assignments`, `unused_mut`,
+   `unused_attributes`, and `dead_code`. Small examples often trigger
+   these lints.
+3. If the example does not contain `extern crate`, then `extern crate
+   <mycrate>;` is inserted (note the lack of `#[macro_use]`).
+4. Finally, if the example does not contain `fn main`, the remainder of the
+   text is wrapped in `fn main() { your_code }`.
+
+For more about that caveat in rule 3, see "Documeting Macros" below.
+
+## Hiding portions of the example
+
+Sometimes, you need some setup code, or other things that would distract
+from your example, but are important to make the tests work. Consider
+an example block that looks like this:
+
+```text
+/// Some documentation.
+# fn foo() {}
+```
+
+It will render like this:
+
+```rust
+/// Some documentation.
+# fn foo() {}
+```
```

Comment:
This one really threw me for a loop just reading the source here before i realized that putting different tags on the code blocks triggers the "hide lines with `#`" behavior `>_>`